CI release notes 788xx (#1059)
harness/developer-hub: release-notes/continuous-integration.md

Additionally, the release notes below are only for NextGen SaaS. FirstGen SaaS release notes are available [here](/docs/first-gen/firstgen-release-notes/harness-saa-s-release-notes) and Self-Managed Enterprise Edition release notes are available [here](/release-notes/self-managed-enterprise-edition).
:::

## March 23, 2023, version 788xx

### What's new

This release does not include new features.

### Early access

This release does not include early access features.

### Fixed issues

> * You can now use [variable expressions](/docs/platform/references/runtime-inputs/#expressions) for the **Shell** and **Image Pull Policy** settings. Previously, selecting the **Expression** input type for **Shell** caused a UI-breaking error, and providing an expression for **Image Pull Policy** caused a field validation error. (CI-7071, ZD-40277)

Reviewer: Do you want to split UI to user interface breaking? I am not sure.

> * If a CI pipeline fails at the **Initialize** step due to an [Azure Repos connector](/docs/platform/connectors/connect-to-a-azure-repo/) having an on-premises **Azure Repos Project URL**, the error message for this failure is more clear as to the cause of the failure. This failure occurs because CI doesn't support Azure DevOps Server Repositories (also known as _on-premises Azure Repos_). (CI-6322)

Reviewer: If a CI pipeline fails at the **Initialize** step due to an [Azure Repos connector](/docs/platform/connectors/connect-to-a-azure-repo/) having an on-premise **Azure Repos Project URL**, the error message for this failure is clear as to the cause of the failure.

> * When configuring a [code repo connector](/docs/category/code-repo-connectors), if the **URL Type** is **Account**, the **Connection Test** now shows the full test repo URL, which is the URL actually used to test the connection, rather than only the account URL. (CI-4398)

Reviewer: When configuring a [code repo connector](/docs/category/code-repo-connectors), if the **URL Type** is **Account**, the **Connection Test** now shows the full test repository URL, which is the URL used to test the connection, rather than the account URL.

Additionally, the release notes below are only for NextGen SaaS. FirstGen SaaS release notes are available [here](/docs/first-gen/firstgen-release-notes/harness-saa-s-release-notes) and Self-Managed Enterprise Edition release notes are available [here](/release-notes/self-managed-enterprise-edition).
:::

## March 24, 2023, version 788xx

### What's new

> * [Cache Intelligence](/docs/continuous-integration/use-ci/caching-ci-data/cache-intelligence/) is now generally available. With Cache Intelligence, Harness automatically caches and restores common dependencies. You don't need to bring your own storage because we store the cache in our hosted environment, Harness Cloud. (CI-7127)

Reviewer: Who is we? Harness? or user?

* [Harness Cloud](/docs/continuous-integration/use-ci/set-up-build-infrastructure/use-harness-cloud-build-infrastructure) build infrastructure now supports **Run as User** for [Plugin](/docs/continuous-integration/ci-technical-reference/plugin-step-settings-reference) and [Run](/docs/continuous-integration/ci-technical-reference/run-step-settings) steps that are running on containers. (CI-7320)

### Early access

This release does not include early access features.

### Fixed issues

* You can now use [variable expressions](/docs/platform/references/runtime-inputs/#expressions) for the **Shell** and **Image Pull Policy** settings. Previously, selecting the **Expression** input type for **Shell** caused a UI-breaking error, and providing an expression for **Image Pull Policy** caused a field validation error. (CI-7071, ZD-40277)

> * If a CI pipeline fails at the **Initialize** step due to an [Azure Repos connector](/docs/platform/connectors/connect-to-a-azure-repo/) having an on-premises **Azure Repos Project URL**, the error message for this failure is clear as to the cause of the failure. This failure occurs because CI doesn't support Azure DevOps Server Repositories (also known as _on-premises Azure Repos_). (CI-6322)

Reviewer 2: ```suggestion
* If a CI pipeline fails at the **Initialize** step due to an [Azure Repos connector](/docs/platform/connectors/connect-to-a-azure-repo/) having an on-premises **Azure Repos Project URL**, the error message clearly describes the root cause. This failure occurs because CI doesn't support Azure DevOps Server Repositories (also known as _on-premises Azure Repos_). (CI-6322)
```

> * If you configure a [code repo connector](/docs/category/code-repo-connectors) where the **URL Type** is **Account**, the **Connection Test** now shows the full test repo URL, which is the URL used to test the connection, rather than the account URL. (CI-4398)

Reviewer: test repository URL

> * When configuring a [code repo connector](/docs/category/code-repo-connectors), if the **URL Type** is **Account**, the **Connection Test** now shows the full test repo URL, which is the URL used to test the connection, rather than the account URL. (CI-4398)

Reviewer 2: ```suggestion
* If you configure a [code repo connector](/docs/category/code-repo-connectors) where the **URL Type** is **Account**, the **Connection Test** now shows the full test repo URL, which is the URL used to test the connection, rather than the account URL. (CI-4398)
```

> * On the [Build details page](/docs/continuous-integration/use-ci/view-your-builds/viewing-builds), the point at which a step displays a successful status is now in sync with the actual time that the step took to succeed. Previously, steps were taking longer to report success than they actually took to run. (CI-7316, ZD-41082)

Reviewer: Previously, steps took longer to report success than they actually took to run.
